src/utils.rs

```diff
@@ -34,6 +35,13 @@ pub fn add_contract_to_ledger_entries(
     Ok(())
 }
 
+pub fn contract_id_from_str(contract_id: &String) -> Result<[u8; 32], FromHexError> {
+    let mut decoded = [0u8; 32];
+    let padded = format!("{:0>width$}", contract_id, width = decoded.len() * 2);
+    hex::decode_to_slice(padded, &mut decoded)?;
+    Ok(decoded)
```

Comment:
Thanks @2opremio! I ended up making a small tweak to this to get it merged because I wanted to use it in an example today.

The change is 50e8191a5f2847b9f1d2c218f733c625f0108fe7. Instead of right padding, it left pads. I realize we originally asked for right-fill (https://github.com/stellar/soroban-cli/pull/43#discussion_r926161225), but after using it I discovered that is problematic if I run contract `1` and then contract `10`. The change also makes it simpler to support single character values like `1`, `2`, etc.

The `format!` functions input is a bit of a monster, but the `0` means pad with zeros, the `>` means right pad, and the `width$` says to get the length from the `width` value.

Reply:
Doesn’t this allow ids longer than 32 bytes to be provided?

Reply:
The `hex::decode_to_slice` function will error with `FromHex::InvalidStringLength` if the hex to decode won't fit into the 32-byte `decoded` slice being passed in.

Reply:
Ah, true, good then